src/Workspaces/Core/Portable/EmbeddedLanguages/RegularExpressions/RegexParser.cs

```diff
@@ -0,0 +1,1927 @@
+                {
+                    components.Add(left);
+                    components.Add(ParseCharacterClassSubtractionNode(minusToken));
+                }
+                else
+                {
+                    var right = ParseRightSideOfCharacterClassRange();
+
+                    if (TryGetRangeComponentValue(left, isRight: false, out var leftCh) &&
+                        TryGetRangeComponentValue(right, isRight: true, out var rightCh) &&
+                        leftCh > rightCh)
+                    {
+                        minusToken = minusToken.AddDiagnosticIfNone(new EmbeddedDiagnostic(
+                            WorkspacesResources.x_y_range_in_reverse_order,
+                            minusToken.GetSpan()));
+                    }
+
+                    components.Add(new RegexCharacterClassRangeNode(left, minusToken, right));
+                }
+            }
+            else
+            {
+                components.Add(left);
+            }
+        }
+
+        private bool IsEscapedMinus(RegexNode node)
+            => node is RegexSimpleEscapeNode simple && IsTextChar(simple.TypeToken, '-');
+
+        private bool TryGetRangeComponentValue(RegexExpressionNode component, bool isRight, out char ch)
+        {
+            // Don't bother examining the component if it has any errors already.  This also means
+            // we don't have to worry about running into invalid escape sequences and the like.
+            if (!HasProblem(component))
+            {
+                return TryGetRangeComponentValueWorker(component, out ch);
+            }
+
+            ch = default;
+            return false;
+        }
+
+        private bool TryGetRangeComponentValueWorker(RegexNode component, out char ch)
+        {
+            switch (component.Kind)
+            {
+                case RegexKind.SimpleEscape:
+                    ch = ((RegexSimpleEscapeNode)component).TypeToken.VirtualChars[0];
+                    switch (ch)
+                    {
+                        case 'a': ch = '\u0007'; break;
+                        case 'b': ch =  '\b'; break;
+                        case 'e': ch = '\u001B'; break;
+                        case 'f': ch = '\f'; break;
+                        case 'n': ch = '\n'; break;
+                        case 'r': ch = '\r'; break;
+                        case 't': ch = '\t'; break;
+                        case 'v': ch = '\u000B'; break;
+                    }
+                    return true;
+
+                case RegexKind.ControlEscape:
+                    var controlEscape = (RegexControlEscapeNode)component;
+                    var controlCh = controlEscape.ControlToken.VirtualChars[0].Char;
+                    // \ca interpreted as \cA
+                    if (controlCh >= 'a' && controlCh <= 'z')
+                    {
+                        controlCh -= (char)('a' - 'A');
+                    }
+                    ch = (char)(controlCh - '@');
```

Comment:
This could use a comment too. I see that `@` is the character before `A`, but that still doesn't quite make sense.

Reply:
I've added the comment and removed usage of '@'   #resolved